### ISSUE TITLE
Pre-training for entity disambiguation

### ISSUE THREAD

**Issue by bennigeir** [studio-ousia/luke#115]
Hi,

I have pretrained an Icelandic version of LUKE, following these [instructions](https://github.com/studio-ousia/luke/issues/41), with good results.
Thank you again for your work and making the code available!

Now I'm very interested in pretraining a model similar to "luke_large_ed.tar.gz" in your [repo](https://github.com/studio-ousia/luke/blob/aa234d934527eda96a9a4c3bd9c7999c05c3fd58/examples/entity_disambiguation/README.md#reproducing-experiments). 

Is this model any different from "luke_base_500k.tar.gz" when it comes to the pretraining phase?
Can I use the instructions provided in [issue 41](https://github.com/studio-ousia/luke/issues/41) to pretrain an ED model? If not, what steps, if any, do I need to take to do so?

**Comment by ikuyamada** [comment]
Hi @bennigeir,

Thank you very much for your interest in LUKE! We trained the entity disambiguation model using the same code available in the repository. We use `bert-large` and hyperparameters listed in the appendix of [our paper](https://arxiv.org/abs/1909.00426). Please note that `--masked-lm-prob` must be set to 0 since we do not use masked language model in the training.

**Comment by bennigeir** [comment]
Thank you @ikuyamada, looks like I'm back on track.

**Comment by bennigeir** [comment]
Hi @ikuyamada,

I have pre-trained an Icelandic version of LUKE and LUKE for ED. The first model works well, however the one for ED unfortunately does not doing as good. I have been following both your paper and information in this repository,

I've tried a few things and thought if you could you confirm that I'm using the "correct" arguments for pretraining the model, or if you see any mistakes?

```
Starting pretraining with the following arguments: {
  "adam_b1": 0.9,
  "adam_b2": 0.999,
  "adam_eps": 1e-06,
  "amp_file": null,
  "batch_size": 8,
  "bert_model_name": "mideind/IceBERT-igc",
  "cpu": false,
  "dataset_dir": "./is-wikipedia-pretrain-dataset/",
  "entity_emb_size": 768,
  "fix_bert_weights": false,
  "fp16": false,
  "fp16_master_weights": true,
  "fp16_max_loss_scale": 4,
  "fp16_min_loss_scale": 1,
  "fp16_opt_level": "O2",
  "global_step": 0,
  "grad_avg_on_cpu": false,
  "gradient_accumulation_steps": 1,
  "learning_rate": 0.0005,
  "local_rank": -1,
  "log_dir": "./logs/ed-luke-icebert-base-768/",
  "lr_schedule": "warmup_linear",
  "mask_words_in_entity_span": false,
  "masked_entity_prob": 0.15,
  "masked_lm_prob": 0.0,
  "master_addr": "127.0.0.1",
  "master_port": "29502",
  "max_grad_norm": 0.0,
  "model_file": null,
  "multilingual": false,
  "node_rank": 0,
  "num_epochs": 1,
  "num_nodes": 1,
  "optimizer_file": null,
  "output_dir": "./ed-luke-icebert-base-768/",
  "parallel": false,
  "random_entity_prob": 0.0,
  "random_word_prob": 0.1,
  "sampling_smoothing": 0.7,
  "save_interval_sec": null,
  "save_interval_steps": null,
  "scheduler_file": null,
  "unmasked_entity_prob": 0.0,
  "unmasked_word_prob": 0.1,
  "warmup_steps": 40,
  "weight_decay": 0.01,
  "whole_word_masking": true
}
```

Thank you in advance!

**Comment by ikuyamada** [comment]
Hi @bennigeir,

Thank you for your continued interest in LUKE :)

We used two-stage training in our experiments. In the first-stage training, we fixed the pretrained BERT parameters by setting `fix_bert_weights` to `True` and trained the model using higher learning rate (i.e., 5e-4) for one epoch. Then, we trained the model by updating all the parameters with smaller learning rate (5e-5) for six epochs. At the beginning of the second-stage training, the model weight was loaded by setting the checkpoint file of the first-stage training to `model_file`.  
We also use a larger batch size (2048) by using gradient accumulation.
